language_server/src/request_manager.cpp

```diff
@@ -64,12 +70,8 @@ void request_manager::end_worker()
 
 bool hlasm_plugin::language_server::request_manager::is_running() 
 {
-    bool result = false;
-    {
-        std::unique_lock<std::mutex> lock(q_mtx_);
-        result = !requests_.empty();
-    }
-    return result;
+    std::lock_guard<std::mutex> lock(q_mtx_);
```

Comment:
i think that this lock is redundant. the whole method seems suspicious. ideally, you want to put lock in some if statement rather then in reading bool

```
{
std::lock_guard<std::mutex> lock(q_mtx_);
if (is_running())
   ...
}
```